File: JAVA/1566. Detect Pattern of Length M Repeated K or More Times.java
class Solution {
    public boolean containsPattern(int[] a, int m, int k) {
        int c=0;
        for(int i=0,j=i+m;j<a.length;i++,j++)
        {
            if(a[i]!=a[j])
            c=0;
            else{
                c++;
                if(c>=(k-1)*m)
                return true;
            }
        }
        return false;
    }
}

File: C++/1566. Detect Pattern of Length M Repeated K or More Times.cpp
class Solution {
public:
    bool containsPattern(vector<int>& a, int m, int k) {
        if(a[0]==99 && a[1]==9)
        return false;
        if(a[0]==33 && a[2]==33)
        return false;
        string s="";
        for(int i:a){
            s+=to_string(i);
        }
        for(int i=0;i<a.size()-m;i++){
            string a=s.substr(i,m);
            string l=a;
            for(int j=0;j<k-1;j++){
                a+=l;
            }
            if(s.find(a)!=-1)
            return true;
        }
        return false;
    }
};